### DISCUSSION TITLE
Save knowledge check responses locally

### DISCUSSION THREAD

**Discussion by ghost** [TheOdinProject/theodinproject#3772] [Ideas]
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this issue follows the `brief description of request` format, e.g. `Add dark mode to website`

<hr>

**1. Description of the Feature Request:**

Each 'Knowledge Check' question would be paired with a textbox for a response, which would be saved locally (e.g. localStorage). Textboxes incentivize articulate responses in words, rather than passive mental reviews, and tracking these responses might be helpful (and potentially motivating) for beginners to monitor their progress. 

Why should these responses be saved? Old answers are insightful to compare former ways of thinking with new ones.

**2. Acceptance Criteria:**
- [ ] Add a textbox to each question
- [ ] Save the response (textbox value) locally

**3. Additional Information:**
Ignore the "Copy" button below. See: https://github.com/TheOdinProject/curriculum/discussions/25394
![demo](https://user-images.githubusercontent.com/102918860/234438338-4c20260d-8de7-4113-9bca-c2bb9a9cbcd4.gif)

**Comment by CouchofTomato** [comment]
Thanks @Raymond-Adams 

This is something I'll need to discuss with the team.

**Comment by KevinMulhern** [comment]
Thanks for opening this @Raymond-Adams, we moved it to a discussion so it can be fleshed out more.
